test/module/irohad/torii/torii_transport_command_test.cpp

```diff
@@ -0,0 +1,290 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <algorithm>
+#include <iterator>
+#include <string>
+
+#include "backend/protobuf/proto_transport_factory.hpp"
+#include "backend/protobuf/proto_tx_status_factory.hpp"
+#include "builders/protobuf/transaction.hpp"
+#include "endpoint.pb.h"
+#include "interfaces/iroha_internal/transaction_batch_factory_impl.hpp"
+#include "interfaces/iroha_internal/transaction_batch_parser_impl.hpp"
+#include "module/irohad/ametsuchi/ametsuchi_mocks.hpp"
+#include "module/irohad/torii/torii_mocks.hpp"
+#include "module/shared_model/builders/protobuf/test_transaction_builder.hpp"
+#include "module/vendor/grpc_mocks.hpp"
+#include "torii/impl/command_service_impl.hpp"
+#include "torii/impl/command_service_transport_grpc.hpp"
+#include "torii/impl/status_bus_impl.hpp"
+#include "torii/processor/transaction_processor_impl.hpp"
+
+constexpr size_t kTimes = 5;
+
+using ::testing::_;
+using ::testing::Invoke;
+using ::testing::Return;
+
+using namespace iroha::network;
+using namespace iroha::ametsuchi;
+using namespace iroha::torii;
+using namespace iroha::synchronizer;
+
+using namespace std::chrono_literals;
+constexpr std::chrono::milliseconds initial_timeout = 1s;
+constexpr std::chrono::milliseconds nonfinal_timeout = 2 * 10s;
+
+class ToriiTransportCommandTest : public testing::Test {
+ public:
+  void SetUp() override {
+    block_query = std::make_shared<MockBlockQuery>();
+    storage = std::make_shared<MockStorage>();
+
+    EXPECT_CALL(*block_query, getTxByHashSync(_))
+        .WillRepeatedly(Return(boost::none));
+    EXPECT_CALL(*storage, getBlockQuery()).WillRepeatedly(Return(block_query));
+
+    status_bus = std::make_shared<MockStatusBus>();
+
+    status_factory =
```

Comment:
I think it could be better to move real dependencies (which are not mocks) move to separate initialization method and describe here why those dependencies are not mocks.